File: code/sorting/src/insertion_sort/insertion_sort.php
<?php
// Part of Cosmos by OpenGenus Foundation
$array = array(0,1,6,7,6,3,4,2);

function insert_sort($array)
{
  for($i = 0; $i < count($array); $i++)
  {
    $val = $array[$i];
    $j = $i-1;
    while($j >= 0 && $array[$j] > $val)
    {
      $array[$j+1] = $array[$j];
      $j--;
    }
    $array[$j+1] = $val;
  }
  return $array;
}

print 'unsorted array';
echo "\n";
print_r($array);
print 'sorted array';
echo "\n";
print_r(insert_sort($array));

?>

File: code/sorting/src/insertion_sort/insertion_sort.py
"""
Part of Cosmos by OpenGenus Foundation

Program for InsertionSort sorting

L :- List/Array of integers
order :- Sorting Order('asc' for Ascending; 'desc' for Descending)
"""


def insertion_sort(L, order):
    length = len(L)
    if order == 'asc':
        for i in range(length):
            key = L[i]
            j = i - 1
            while j >= 0 and key < L[j]:
                L[j + 1] = L[j]
                j -= 1
            L[j + 1] = key
        return L
    elif order == 'desc':
        for i in range(length):
            key = L[i]
            j = i - 1
            while j >= 0 and key > L[j]:
                L[j + 1] = L[j]
                j -= 1
            L[j + 1] = key
        return L


if __name__ == "__main__":

    # creating a random array of length 100
    L = np.random.random_integers(100, size=(100,)).tolist()
    print("Given array : \n", L)
    print("Sorted array : \n", insertion_sort(L, 'asc'))
